### DISCUSSION TITLE
Cannot get the API working correctly - TypeError: '>' not supported between instances of 'str' and 'int'

### DISCUSSION THREAD

**Discussion by Teedo145** [lllyasviel/Fooocus#1512] [Q&A]
I am using code that apparently works for others but can't fix this error

Traceback (most recent call last):
  File "D:\Fooocus\Fooocus\modules\async_worker.py", line 803, in worker
    handler(task)
  File "D:\Fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\Fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\Fooocus\Fooocus\modules\async_worker.py", line 324, in handler
    if advanced_parameters.overwrite_switch > 0:
TypeError: '>' not supported between instances of 'str' and 'int'
Total time: 0.08 seconds

from gradio_client import Client

client = Client("http://127.0.0.1:7865/", serialize=False)
result = client.predict(
    "hello!",  # str in 'parameter_10' Textbox component
    "hey!",  # str in 'Negative Prompt' Textbox component
    ["Fooocus V2"],  # List[str] in 'Selected Styles' Checkboxgroup component
    "Speed",  # str in 'Performance' Radio component
    "1024×1024 <span style=\"color: grey;\"> ∣ 1:1</span>",  # str in 'Aspect Ratios' Radio component
    1,  # int | float (numeric value between 1 and 32) in 'Image Number' Slider component
    "123",  # str in 'Seed' Textbox component
    4,  # int | float (numeric value between 0.0 and 30.0) in 'Image Sharpness' Slider component
    4,  # int | float (numeric value between 1.0 and 30.0) in 'Guidance Scale' Slider component
    "juggernautXL_version6Rundiffusion.safetensors", # str (Option from: ['colossusProjectXLSFW_v53Trained.safetensors', 'formulaxlXLComfyui_v20Pruned.safetensors', 'juggernautXL_version6Rundiffusion.safetensors', 'realismEngineSDXL_v20VAE.safetensors', 'realisticStockPhoto_v10.safetensors'])	in 'Base Model (SDXL only)' Dropdown component
    "None", # str (Option from: ['None', 'colossusProjectXLSFW_v53Trained.safetensors', 'formulaxlXLComfyui_v20Pruned.safetensors', 'juggernautXL_version6Rundiffusion.safetensors', 'realismEngineSDXL_v20VAE.safetensors', 'realisticStockPhoto_v10.safetensors']) in 'Refiner (SDXL or SD 1.5)' Dropdown component
    0.5,  # int | float (numeric value between 0.1 and 1.0) in 'Refiner Switch At' Slider component
    "None", # str (Option from: ['None', 'SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 'sdxl_lcm_lora.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 'SOAP_LoRA.safetensors']) in 'LoRA 1' Dropdown component
    -2,  # int | float (numeric value between -2 and 2)	in 'Weight' Slider component
    "None", # str (Option from: ['None', 'SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 'sdxl_lcm_lora.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 'SOAP_LoRA.safetensors']) in 'LoRA 2' Dropdown component
    -2,  # int | float (numeric value between -2 and 2) in 'Weight' Slider component
    "None", # str (Option from: ['None', 'SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 'sdxl_lcm_lora.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 'SOAP_LoRA.safetensors']) in 'LoRA 3' Dropdown component
    -2,  # int | float (numeric value between -2 and 2) in 'Weight' Slider component
    "None", # str (Option from: ['None', 'SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 'sdxl_lcm_lora.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 'SOAP_LoRA.safetensors']) in 'LoRA 4' Dropdown component
    -2,  # int | float (numeric value between -2 and 2) in 'Weight' Slider component
    "None", # str (Option from: ['None', 'SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 'sdxl_lcm_lora.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 'SOAP_LoRA.safetensors']) in 'LoRA 5' Dropdown component
    -2,  # int | float (numeric value between -2 and 2) in 'Weight' Slider component
    True,  # bool in 'Input Image' Checkbox component
    "heythere!",  # str in 'parameter_82' Textbox component
    "Disabled",  # str in 'Upscale or Variation:' Radio component
    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image)in 'Drag above image to here' Image component
    ["Left"],  # List[str] in 'Outpaint Direction' Checkboxgroup component
    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image) in 'Drag above image to here' Image component
    "heyy5!",  # str in 'Inpaint Additional Prompt' Textbox component
    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image) in 'Image' Image component
    0,  # int | float (numeric value between 0.0 and 1.0) in 'Stop At' Slider component
    0,  # int | float (numeric value between 0.0 and 2.0) in 'Weight' Slider component
    "ImagePrompt",  # str in 'Type' Radio component
    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image) in 'Image' Image component
    0,  # int | float (numeric value between 0.0 and 1.0) in 'Stop At' Slider component
    0,  # int | float (numeric value between 0.0 and 2.0) in 'Weight' Slider component
    "ImagePrompt",  # str in 'Type' Radio component
    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image) in 'Image' Image component
    0,  # int | float (numeric value between 0.0 and 1.0) in 'Stop At' Slider component
    0,  # int | float (numeric value between 0.0 and 2.0) in 'Weight' Slider component
    "ImagePrompt",  # str in 'Type' Radio component
    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image) in 'Image' Image component
    0,  # int | float (numeric value between 0.0 and 1.0) in 'Stop At' Slider component
    0,  # int | float (numeric value between 0.0 and 2.0) in 'Weight' Slider component
    "ImagePrompt",  # str in 'Type' Radio component
    fn_index=30
)
print(result)

**Comment by psydox** [comment]
> I am using code that apparently works for others but can't fix this error
> 
> Traceback (most recent call last): File "D:\Fooocus\Fooocus\modules\async_worker.py", line 803, in worker handler(task) File "D:\Fooocus\python_embeded\lib\site-packages\torch\utils_contextlib.py", line 115, in decorate_context return func(*args, **kwargs) File "D:\Fooocus\python_embeded\lib\site-packages\torch\utils_contextlib.py", line 115, in decorate_context return func(*args, **kwargs) File "D:\Fooocus\Fooocus\modules\async_worker.py", line 324, in handler if advanced_parameters.overwrite_switch > 0: TypeError: '>' not supported between instances of 'str' and 'int' Total time: 0.08 seconds
> 
> from gradio_client import Client
> 
> client = Client("http://127.0.0.1:7865/", serialize=False) result = client.predict( "hello!", # str in 'parameter_10' Textbox component "hey!", # str in 'Negative Prompt' Textbox component ["Fooocus V2"], # List[str] in 'Selected Styles' Checkboxgroup component "Speed", # str in 'Performance' Radio component "1024×1024 <span style="color: grey;"> ∣ 1:1", # str in 'Aspect Ratios' Radio component 1, # int | float (numeric value between 1 and 32) in 'Image Number' Slider component "123", # str in 'Seed' Textbox component 4, # int | float (numeric value between 0.0 and 30.0) in 'Image Sharpness' Slider component 4, # int | float (numeric value between 1.0 and 30.0) in 'Guidance Scale' Slider component "juggernautXL_version6Rundiffusion.safetensors", # str (Option from: ['colossusProjectXLSFW_v53Trained.safetensors', 'formulaxlXLComfyui_v20Pruned.safetensors', 'juggernautXL_version6Rundiffusion.safetensors', 'realismEngineSDXL_v20VAE.safetensors', 'realisticStockPhoto_v10.safetensors']) in 'Base Model (SDXL only)' Dropdown component "None", # str (Option from: ['None', 'colossusProjectXLSFW_v53Trained.safetensors', 'formulaxlXLComfyui_v20Pruned.safetensors', 'juggernautXL_version6Rundiffusion.safetensors', 'realismEngineSDXL_v20VAE.safetensors', 'realisticStockPhoto_v10.safetensors']) in 'Refiner (SDXL or SD 1.5)' Dropdown component 0.5, # int | float (numeric value between 0.1 and 1.0) in 'Refiner Switch At' Slider component "None", # str (Option from: ['None', 'SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 'sdxl_lcm_lora.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 'SOAP_LoRA.safetensors']) in 'LoRA 1' Dropdown component -2, # int | float (numeric value between -2 and 2) in 'Weight' Slider component "None", # str (Option from: ['None', 'SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 'sdxl_lcm_lora.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 'SOAP_LoRA.safetensors']) in 'LoRA 2' Dropdown component -2, # int | float (numeric value between -2 and 2) in 'Weight' Slider component "None", # str (Option from: ['None', 'SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 'sdxl_lcm_lora.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 'SOAP_LoRA.safetensors']) in 'LoRA 3' Dropdown component -2, # int | float (numeric value between -2 and 2) in 'Weight' Slider component "None", # str (Option from: ['None', 'SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 'sdxl_lcm_lora.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 'SOAP_LoRA.safetensors']) in 'LoRA 4' Dropdown component -2, # int | float (numeric value between -2 and 2) in 'Weight' Slider component "None", # str (Option from: ['None', 'SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 'sdxl_lcm_lora.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 'SOAP_LoRA.safetensors']) in 'LoRA 5' Dropdown component -2, # int | float (numeric value between -2 and 2) in 'Weight' Slider component True, # bool in 'Input Image' Checkbox component "heythere!", # str in 'parameter_82' Textbox component "Disabled", # str in 'Upscale or Variation:' Radio component "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image)in 'Drag above image to here' Image component ["Left"], # List[str] in 'Outpaint Direction' Checkboxgroup component "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image) in 'Drag above image to here' Image component "heyy5!", # str in 'Inpaint Additional Prompt' Textbox component "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image) in 'Image' Image component 0, # int | float (numeric value between 0.0 and 1.0) in 'Stop At' Slider component 0, # int | float (numeric value between 0.0 and 2.0) in 'Weight' Slider component "ImagePrompt", # str in 'Type' Radio component "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image) in 'Image' Image component 0, # int | float (numeric value between 0.0 and 1.0) in 'Stop At' Slider component 0, # int | float (numeric value between 0.0 and 2.0) in 'Weight' Slider component "ImagePrompt", # str in 'Type' Radio component "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image) in 'Image' Image component 0, # int | float (numeric value between 0.0 and 1.0) in 'Stop At' Slider component 0, # int | float (numeric value between 0.0 and 2.0) in 'Weight' Slider component "ImagePrompt", # str in 'Type' Radio component "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVQImWNgYGBgAAAABQABh6FO1AAAAABJRU5ErkJggg==", # str (filepath or URL to image) in 'Image' Image component 0, # int | float (numeric value between 0.0 and 1.0) in 'Stop At' Slider component 0, # int | float (numeric value between 0.0 and 2.0) in 'Weight' Slider component "ImagePrompt", # str in 'Type' Radio component fn_index=30 ) print(result)

I had this issue as well before.  I found out that the "seed" should be "INT" and not "STR".  Try adding numbers instead.  Remove the double quote.

**Reply by Teedo145** [reply]
Thanks, unfortunately I have tried that and it still has the same error :(
I tried changing all of the possible integers that are strings to integers

**Reply by psydox** [reply]
I copied your code and managed to run it on my system.  

Try escaping the double-quote character that sandwich the color: grey; on the aspect ratio.

It should be:
"1024×1024 <span style=\\"color: grey;\\"> ∣ 1:1", # str in 'Aspect Ratios' Radio component

**Reply by Teedo145** [reply]
That's super strange...unfortunately it already is escaped but it didn't seem to show it on here :(
